docs/rfcs/2023-12-22-enclose-column-id.md

```diff
@@ -0,0 +1,42 @@
+---
+Feature Name: Enclose Region Id
+Tracking Issue: https://github.com/GreptimeTeam/greptimedb/issues/2982
+Date: 2023-12-22
+Author: "Ruihang Xia <waynestxia@gmail.com>"
+---
+
+# Summary
+This RFC proposes to enclose the usage of `RegionId` into the region engine only.
+
+# Motivation
+`ColumnId` is an identifier for columns. It's assigned by meta server, stored in `TableInfo` and `RegionMetadata` and used in region engine to distinguish columns.
+
+At present, Both Frontend, Datanode and Metasrv are aware of `ColumnId` but it's only used in region engine. Thus this RFC proposes to remove it from Frontend (mainly used in `TableInfo`) and Metasrv.
+
+# Details
+
+`RegionId` is used widely on both read and write paths. Removing it from Frontend and Metasrv implies several things:
+
+- A column may have different column id in different regions.
```

Comment:
For simplicity, we can still assume columns have the same column ids across regions. The procedure maintains consistency between regions and the region engine should ensure alterations are idempotent.

Reply:
Thanks for your kind review. I've added this paragraph to the explanation.